### ISSUE TITLE
Panic error on block producer

### ISSUE THREAD

**Issue by MattKetmo** [berachain/beacon-kit#1422]
Hello, I notice each time my `beacond` is a block producer, it results in the following error (here for block height 118367):

```
ERR panic recovered in PrepareProposal height=118367 module=baseapp panic="runtime error: cannot convert slice with length 64 to array or pointer to array with length 96"
```

- beacond version `v0.2.0-alpha.0` (from github release or self compiled) 
- run on alpine or debian, always the same error
- geth version `v1.14.5`

Also, I'm not receiving any EL fees, I don't know if this issue is the cause or a consequence.

**Comment by MattKetmo** [comment]
More context on running processes:

### Peers

- beacon-kit = 40 peers
- geth = 13 outbound peers, 5 inbound peers


### Config
geth config:
```
geth
  --port=40404
  --authrpc.jwtsecret=/jwt-secret/jwt
  --authrpc.addr=0.0.0.0
  --authrpc.vhosts=*
  --datadir=/root/.ethereum
  --http
  --http.addr=0.0.0.0
  --http.api=eth,net,web3,txpool,debug
  --http.corsdomain=*
  --http.port=8545
  --http.vhosts=*
  --metrics
  --metrics.addr=0.0.0.0
  --ws
  --ws.addr=0.0.0.0
  --ws.api=eth,net,web3
  --ws.origins=*
  --ws.port=8545
  --bootnodes=enode://0401e494dbd0c84c5c0f72adac5985d2f2525e08b68d448958aae218f5ac8198a80d1498e0ebec2ce38b1b18d6750f6e61a56b4614c5a6c6cf0981c39aed47dc@34.159.32.127:30303,enode://...
  --discovery.port=40404
  --networkid=80084
  --snapshot=false
  --syncmode=snap
```

beacon-kit config = the default files generated by `beacond init` + these override (to match the [files of the networks folder](https://github.com/berachain/beacon-kit/tree/main/testing/networks/80084) + custom value like fee recipient & rpc endpoint)
```
app.toml:
  pruning: "nothing"
  api:
    enable: true
    enabled-unsafe-cors: true
    swagger: true
  beacon-kit:
    engine:
      jwt-secret-path: '/jwt-secret/jwt'
      rpc-dial-url: 'http://xxxx:8551'
    kzg:
      trusted-setup-path: '/home/app/.beacond/config/kzg-trusted-setup.json'
    payload-builder:
      suggested-fee-recipient: '0x64FE2F0C072858A9742917e48e29BD939EeBB179'
      payload-timeout: "1.2s"
    validator:
      # enable-optimistic-payload-builds: "false"
config.toml:
  log_format: "plain"
  log_level: "info"
  consensus:
    double_sign_check_height: 5
    timeout_commit: "1.2s"
    timeout_propose: "1.7s"
  mempool:
    type: "nop"
    recheck_timeout: "0s"
    cache_size: 0
  p2p:
    max_num_inbound_peers: 100
    max_num_outbound_peers: 40
```

### Logs

geth
```
...
INFO [06-11|13:00:17.488] Imported new potential chain segment     number=134,048 hash=879b13..adf4d8 blocks=1  txs=28  mgas=4.601  elapsed=15.033ms    mgasps=306.051  snapdiffs=2.23MiB    triediffs=13.43MiB triedirty=29.69MiB
INFO [06-11|13:00:17.496] Chain head was updated                   number=134,048 hash=879b13..adf4d8 root=ce831e..b874f6 elapsed="766.97µs"
INFO [06-11|13:00:17.497] Starting work on payload                 id=0x039e2a1cffb8583b
INFO [06-11|13:00:17.502] Updated payload                          id=0x039e2a1cffb8583b number=134,049 hash=e85b03..bf506b txs=6   withdrawals=16 gas=927,725    fees=0.0017800875    root=b2d52f..fe8fa4 elapsed=4.925ms
INFO [06-11|13:00:18.795] Stopping work on payload                 id=0x03bbb233a1f39439 reason=timeout
INFO [06-11|13:00:19.462] Imported new potential chain segment     number=134,049 hash=9f2fc8..67114f blocks=1  txs=7   mgas=1.026  elapsed=8.421ms     mgasps=121.878  snapdiffs=2.23MiB    triediffs=13.43MiB triedirty=29.69MiB
INFO [06-11|13:00:19.478] Chain head was updated                   number=134,049 hash=9f2fc8..67114f root=db80b9..19ecb7 elapsed="721.83µs"
INFO [06-11|13:00:19.479] Starting work on payload                 id=0x0381d2577541ef4e
INFO [06-11|13:00:19.481] Updated payload                          id=0x0381d2577541ef4e number=134,050 hash=2cc67e..4521d1 txs=3   withdrawals=16 gas=692,552    fees=0.001683129     root=d034bb..4e4153 elapsed=2.630ms
INFO [06-11|13:00:20.649] Stopping work on payload                 id=0x032eb95f338c86db reason=timeout
...
```

beacond
```
...
1:01PM INF calling new payload is_optimistic=false module=beacon-kit payload_block_hash=0xfda51d0ccd3dc5719cacf585a1f472af0943034d3d3391bae4171cfb08a6dcbe payload_parent_block_hash=0x6f5a94c6374d604583bd2a9dd28b708d030b16c33a7cc3018ab6a6d437d0a885 service=execution-engine
1:01PM INF state root verification succeeded - accepting incoming beacon block 🏎️  module=beacon-kit service=validator state_root=0x07c351a2409dd875c8fe40f8935c69e624c4b9cbf2a0e10066b3d6c163c2c4e2
1:01PM INF optimistically triggering payload build for next slot 🛩️  module=beacon-kit next_slot=0x277f6 service=validator
1:01PM INF notifying forkchoice update finalized_eth1_hash=0x6f5a94c6374d604583bd2a9dd28b708d030b16c33a7cc3018ab6a6d437d0a885 has_attributes=true head_eth1_hash=0xfda51d0ccd3dc5719cacf585a1f472af0943034d3d3391bae4171cfb08a6dcbe module=beacon-kit safe_eth1_hash=0x6f5a94c6374d604583bd2a9dd28b708d030b16c33a7cc3018ab6a6d437d0a885 service=execution-engine
1:01PM INF bob the builder; can we forkchoice update it?; bob the builder; yes we can 🚧 for_slot=0x277f6 head_eth1_hash=0xfda51d0ccd3dc5719cacf585a1f472af0943034d3d3391bae4171cfb08a6dcbe module=beacon-kit parent_block_root=0x8d35feab4b9068da1baa9262a03002a32722208f931e1c676775a49e82b12bb1 payload_id=0x033bdfb959f3109a service=payload-builder
1:01PM INF finalizing commit of block hash=0F3DD151EE64259865D0DD6299AA19271F45DFBC700EDD07A1D0F01256B1F8BA height=161781 module=consensus num_txs=2 root=72786C73AEA2E6785B5DEADE69AA43234496D4BBCB499D922C11131040E8AA4C
1:01PM INF finalized block block_app_hash=4B86D408610267BC7D35FB2AE65DC5077B789894B4464EC369806A78E2AF9FB7 height=161781 module=state num_txs_res=2 num_val_updates=0 syncing_to_height=161781
1:01PM INF committed state block_app_hash=72786C73AEA2E6785B5DEADE69AA43234496D4BBCB499D922C11131040E8AA4C height=161781 module=state
1:01PM INF indexed block events height=161781 module=txindex
1:01PM INF Timed out dur=1093.957922 height=161782 module=consensus round=0 step=RoundStepNewHeight
1:01PM INF received proposal module=consensus proposal="Proposal{161782/0 (FFF102CA89E303017D3EDCD2EDE9FF3C7AF8BD4E53BF472D4922FF0BC41DC4CF:1:7D3BD2563D74, -1) B0ED1AE88582 @ 2024-06-11T13:01:09.026098413Z}" proposer=2EDC26CB83D155007D7DEB5A8F52AB7FA79FC23B
1:01PM INF received complete proposal block hash=FFF102CA89E303017D3EDCD2EDE9FF3C7AF8BD4E53BF472D4922FF0BC41DC4CF height=161782 module=consensus
1:01PM INF received proposal with beacon_block=true blob_sidecars=true module=baseapp service=prepare-proposal
1:01PM INF no blob sidecars to verify, skipping verifier 🧢  module=beacon-kit service=validator slot=0x277f6
1:01PM INF received incoming beacon block 📫  module=beacon-kit service=validator state_root=0xfead258d4907a267b7b81b0bcec025686c551bb24534eefb67bb3d65d8f641a6
...
```

more logs around the **panic error when block proposer**
```
1:01PM INF finalized block block_app_hash=B4771A9B2E8580755AA3ADEDC6257B303B8E4767F3C656D2EBAB7B465116247A height=161796 module=state num_txs_res=2 num_val_updates=0 syncing_to_height=161796
1:01PM INF committed state block_app_hash=B8584D0C41788FA2CC24FF86F2D10C4B2B1AFE95192745F73408908833A9262C height=161796 module=state
1:01PM INF indexed block events height=161796 module=txindex
1:01PM INF Timed out dur=1092.91609 height=161797 module=consensus round=0 step=RoundStepNewHeight
1:01PM INF requesting beacon block assembly 🙈 module=beacon-kit service=validator slot=0x27805
1:01PM ERR panic recovered in PrepareProposal height=161797 module=baseapp panic="runtime error: cannot convert slice with length 64 to array or pointer to array with length 96"
1:01PM INF received proposal module=consensus proposal="Proposal{161797/0 (D9633D9E17D4F3FA5DC94CD7B0518007ECF60198A3E42E4F5DDC9ADE70A1FF33:1:F2FF0BB5A36A, -1) B6558BB3A154 @ 2024-06-11T13:01:44.749158351Z}" proposer=6BDE137EBAE3BF317C8CF1C155F55677C772A40A
1:01PM INF received complete proposal block hash=D9633D9E17D4F3FA5DC94CD7B0518007ECF60198A3E42E4F5DDC9ADE70A1FF33 height=161797 module=consensus
1:01PM INF received proposal with beacon_block=false blob_sidecars=false module=baseapp service=prepare-proposal
1:01PM WRN aborting blob verification - beacon block not found in proposal 🚫  module=beacon-kit service=validator
1:01PM WRN aborting block verification - beacon block not found in proposal 🚫  module=beacon-kit service=validator
1:01PM INF finalizing commit of block hash=D9633D9E17D4F3FA5DC94CD7B0518007ECF60198A3E42E4F5DDC9ADE70A1FF33 height=161797 module=consensus num_txs=0 root=B4771A9B2E8580755AA3ADEDC6257B303B8E4767F3C656D2EBAB7B465116247A
1:01PM INF finalized block block_app_hash=B4771A9B2E8580755AA3ADEDC6257B303B8E4767F3C656D2EBAB7B465116247A height=161797 module=state num_txs_res=0 num_val_updates=0 syncing_to_height=161797
1:01PM INF committed state block_app_hash=B4771A9B2E8580755AA3ADEDC6257B303B8E4767F3C656D2EBAB7B465116247A height=161797 module=state
1:01PM INF indexed block events height=161797 module=txindex
1:01PM INF Timed out dur=1118.178796 height=161798 module=consensus round=0 step=RoundStepNewHeight
1:01PM INF received proposal module=consensus proposal="Proposal{161798/0 (CA3E8843784C8677D1BD4F52D06B0B95815A03A33015DA3B2C5BA00E62C966A3:1:2B6A85864BB7, -1) 8C4DA3E95E7D @ 2024-06-11T13:01:46.558135667Z}" proposer=6E5C56EAEBFF1A0796375FD342961948385DBFD4
1:01PM INF received complete proposal block hash=CA3E8843784C8677D1BD4F52D06B0B95815A03A33015DA3B2C5BA00E62C966A3 height=161798 module=consensus
1:01PM INF received proposal with beacon_block=true blob_sidecars=true module=baseapp service=prepare-proposal
1:01PM INF no blob sidecars to verify, skipping verifier 🧢  module=beacon-kit service=validator slot=0x27806
1:01PM INF received incoming beacon block 📫  module=beacon-kit service=validator state_root=0xd0cf0473fdd7bb432000bf8c02d917a4e44cd18306a8856308b42654b50be117
1:01PM INF calling new payload is_optimistic=false module=beacon-kit payload_block_hash=0xff41c9c0aa867e7086cb2dda1edcc033e4deae1e9b0c5f4e27ce674235b3f862 payload_parent_block_hash=0x77ef27283b8e4ea74f470a05b6a1f5aae7a7931ea95b61fb8e2a5e47f06dedaa service=execution-engine
1:01PM INF state root verification succeeded - accepting incoming beacon block 🏎️  module=beacon-kit service=validator state_root=0xd0cf0473fdd7bb432000bf8c02d917a4e44cd18306a8856308b42654b50be117
1:01PM INF optimistically triggering payload build for next slot 🛩️  module=beacon-kit next_slot=0x27807 service=validator
1:01PM INF notifying forkchoice update finalized_eth1_hash=0x77ef27283b8e4ea74f470a05b6a1f5aae7a7931ea95b61fb8e2a5e47f06dedaa has_attributes=true head_eth1_hash=0xff41c9c0aa867e7086cb2dda1edcc033e4deae1e9b0c5f4e27ce674235b3f862 module=beacon-kit safe_eth1_hash=0x77ef27283b8e4ea74f470a05b6a1f5aae7a7931ea95b61fb8e2a5e47f06dedaa service=execution-engine
1:01PM INF bob the builder; can we forkchoice update it?; bob the builder; yes we can 🚧 for_slot=0x27807 head_eth1_hash=0xff41c9c0aa867e7086cb2dda1edcc033e4deae1e9b0c5f4e27ce674235b3f862 module=beacon-kit parent_block_root=0xb3f22f96077b821c64cbc8483ee4c8aba7a83bc538885b665e95996228f384aa payload_id=0x03a7dfe64151f83b service=payload-builder
1:01PM INF finalizing commit of block hash=CA3E8843784C8677D1BD4F52D06B0B95815A03A33015DA3B2C5BA00E62C966A3 height=161798 module=consensus num_txs=2 root=B4771A9B2E8580755AA3ADEDC6257B303B8E4767F3C656D2EBAB7B465116247A
1:01PM INF finalized block block_app_hash=ACD9096350A4D3C9DDF99F71A74C491E5FF01927040B741749E4F59F94D58DA2 height=161798 module=state num_txs_res=2 num_val_updates=0 syncing_to_height=161798
1:01PM INF committed state block_app_hash=B4771A9B2E8580755AA3ADEDC6257B303B8E4767F3C656D2EBAB7B465116247A height=161798 module=state
1:01PM INF indexed block events height=161798 module=txindex
``

**Comment by itsdevbear** [comment]
hey @MattKetmo thanks for submitting the bug, not receiving fees would be a side affect of never proposing a block.

**Comment by itsdevbear** [comment]
my guess is that your signing setup is setup incorrectly, as 96 bytes is the length of a BLSSignature and you are getting a 64 != 96

**Comment by itsdevbear** [comment]
Can you check your validator keys to ensure they are BLS?

**Comment by MattKetmo** [comment]
@itsdevbear my `priv_validator_key.json` looks like this (replacing values with `x` of the same length)
```
{
  "address": "6BDE137EBAE3BF317C8CF1C155F55677C772A40A",
  "pub_key": {
    "type": "cometbft/PubKeyBls12_381",
    "value": "xxxxxxxxxxx+xxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/"
  },
  "priv_key": {
    "type": "cometbft/PrivKeyBls12_381",
    "value": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx="
  }
}
```

**Comment by itsdevbear** [comment]
hmmmm

I am merging this: https://github.com/berachain/beacon-kit/pull/1432, it won't help your get unstuck, but it will help give a more specific error.

**Comment by itsdevbear** [comment]
I'm fairly certain your signer is not configured correctly.

**Comment by itsdevbear** [comment]
yup it's non state breaking

**Comment by MattKetmo** [comment]
@itsdevbear oh, just found the cause:

Actually I load my validator key from vault, in a memory mount point on `/vault/secrets` (via vault injector)

When I do, I update my `config.toml` like this
```
node_key_file = '/vault/secrets/node_key.json'
priv_validator_key_file = '/vault/secrets/priv_validator_key.json'
```

When beacond starts, he correctly reads that file since it's printing the address in it (with the mention "your node is validator").

However I still have some placeholder files in `.beacond/config/priv_validator_key.json` and `.beacond/config/node_key.json`, which are the issue.

These files are just here to prevent the process from crashing when I start the node without validation keys (without vault injector and without the config edit). So I don't expect `beacond` to read these files when I'm in the validator node. (that's how we operate all our Cosmos chain nodes)

And guess what: `.beacond/config/priv_validator_key.json` wasn't a bls key. 

I copied the files from `/vault/secrets` to `~/.beacond/config/` and it worked 🪄  I got my first rewards.

👉 So, it looks like the default priv_key path is being read somewhere when it shouldn't.
